config/config.go

```diff
@@ -119,53 +127,73 @@ type Nodepool struct {
-	OSDiskSizeGB                        int64    `json:"osDiskSizeGB" yaml:"osDiskSizeGB"`
-	PrivateAccess                       bool     `json:"privateAccess" yaml:"privateAccess"`
-	ProviderVersion                     string   `json:"providerVersion" yaml:"providerVersion"`
-	PublicAccess                        bool     `json:"publicAccess" yaml:"publicAccess"`
-	Region                              string   `json:"region" yaml:"region"`
-	ResourceGroup                       string   `json:"resourceGroup" yaml:"resourceGroup"`
-	ResourceLocation                    string   `json:"resourceLocation" yaml:"resourceLocation"`
-	VMSize                              string   `json:"vmSize" yaml:"vmSize"`
+	Ami                                 string                  `json:"ami,omitempty" yaml:"ami,omitempty"`
+	AvailabilityZones                   []string                `json:"availabilityZones" yaml:"availabilityZones"`
+	AWSAccessKey                        string                  `json:"awsAccessKey" yaml:"awsAccessKey"`
+	AWSInstanceType                     string                  `json:"awsInstanceType" yaml:"awsInstanceType"`
+	AWSRootSize                         int64                   `json:"awsRootSize" yaml:"awsRootSize"`
+	AWSSecretKey                        string                  `json:"awsSecretKey" yaml:"awsSecretKey"`
+	AWSSecurityGroupNames               []string                `json:"awsSecurityGroupNames" yaml:"awsSecurityGroupNames"`
+	AWSSecurityGroups                   []string                `json:"awsSecurityGroups" yaml:"awsSecurityGroups"`
+	AWSSubnetID                         string                  `json:"awsSubnetID" yaml:"awsSubnetID"`
+	AWSSubnets                          []string                `json:"awsSubnets" yaml:"awsSubnets"`
+	AWSVpcID                            string                  `json:"awsVpcID" yaml:"awsVpcID"`
+	AWSZoneLetter                       string                  `json:"awsZoneLetter" yaml:"awsZoneLetter"`
+	AzureClientID                       string                  `json:"azureClientID" yaml:"azureClientID"`
+	AzureClientSecret                   string                  `json:"azureClientSecret" yaml:"azureClientSecret"`
+	AzureSubscriptionID                 string                  `json:"azureSubscriptionID" yaml:"azureSubscriptionID"`
+	CloudCredentialName                 string                  `json:"cloudCredentialName" yaml:"cloudCredentialName"`
+	DefaultClusterRoleForProjectMembers string                  `json:"defaultClusterRoleForProjectMembers" yaml:"defaultClusterRoleForProjectMembers"`
+	EnableNetworkPolicy                 bool                    `json:"enableNetworkPolicy" yaml:"enableNetworkPolicy"`
+	ETCD                                *rkev1.ETCD             `json:"etcd,omitempty" yaml:"etcd,omitempty"`
+	ETCDRKE1                            *management.ETCDService `json:"etcdRKE1,omitempty" yaml:"etcdRKE1,omitempty"`
```

Comment:
nothing else has 'omitempty' should it be rm'd from etcd variables?

Reply:
Resolved